src/uagents/context.py

```diff
@@ -97,6 +99,22 @@ def get_message_protocol(self, message_schema_digest) -> Optional[str]:
                     return protocol_digest
         return None
 
+    def get_agents_by_protocol(self, protocol_digest: str) -> list:
```

Comment:
```suggestion
    def get_agents_by_protocol(self, protocol_digest: str, limit: Optional[int] = None) -> List[str]:
```

The return type should be a bit more specific, also there needs to be a `limit` in the API for the maximum number of entries that are returned.